lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -44,10 +45,12 @@ def build_apps_v1beta1_kubeclient(context)
     end
 
     def _build_kubeclient(api_version:, context:, endpoint_path: nil)
-      config = GoogleFriendlyConfig.read(ENV.fetch("KUBECONFIG"))
-      unless config.contexts.include?(context)
-        raise ContextMissingError, context
-      end
+      # Find a context defined in kube conf files that matches the input context by name
+      friendly_configs = config_files.map { |f| GoogleFriendlyConfig.read(f) }
+      config = friendly_configs.each.find { |c| c.contexts.include?(context) }
```

Comment:
You don't even need the `.each` here. `Array` is an Enumerable, so you can call `.find` on it directly

Reply:
Updated.